### PR TITLE
refactor(pairing): Extract a256gcm related utils into its own module. 

### PR DESCRIPTION
**Pull request by shane-tomlinson**
extraction from #6514

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

File: app/scripts/lib/crypto/a256gcm.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    });
+  },
+
+  /**
+   * encrypts `plaintext` using `keysJwk`
+   *
+   * @param {String} plaintext Data to encrypt
+   * @param {String} keysJwk keysJwk to encrypt data
+   * @param {Object} [options={}] Options
+   *   @param {String} [options.unsafeExplicitIV] - Initialization vector used to create bundle for testing purposes
```
Comment (philbooth) [inline app/scripts/lib/crypto/a256gcm.js]: Excellent name! Is there (or could we add) a lint rule that warns on `unsafeExplicit...` property names in `app/scripts`?

Comment (shane-tomlinson) [reply app/scripts/lib/crypto/a256gcm.js]: @vbudhram deserves the credit, I'm not smart enough to come up with a name like that.

Comment (shane-tomlinson) [reply app/scripts/lib/crypto/a256gcm.js]: @philbooth - mind checking the 2nd commit of this PR? I used grunt-ban-words to check for unsafeExplicitIV. I could do something like [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/no-danger.js) and ban just that word, though that seems like more work.

**Comment by philbooth** [comment]
r+ for the second commit!

(I didn't mean to suggest it for this PR btw, but well done anyway!)

File: app/scripts/lib/crypto/a256gcm.js (hunk 1 above)
Comment (vbudhram) [reply app/scripts/lib/crypto/a256gcm.js]: I can't take credit either, it was inspired by rfk!

**Pull request merged**